corpus/convert.py

```diff
@@ -93,6 +106,59 @@ def run(self):
         c = corpus.Corpus()
         c.load(corpus_path)
 
-        with open(self.txt_path.get_path(), "wt") as out:
-            for segment in c.segments():
+        if self.segment_file:
+            with open(tk.uncached_path(self.segment_file), "rt") as f:
+                lines = [l for l in f.readlines() if len(l.strip()) > 0]
+                segments_whitelist = set(lines)
+        else:
+            segments_whitelist = None
+
+        if self.gzip:
+            out = gzip.open(self.out_txt.get_path(), "wt")
+        else:
+            out = open(self.out_txt.get_path(), "wt")
+
+        for segment in c.segments():
+            if not segments_whitelist or segment.fullname in segments_whitelist:
                 out.write(segment.orth + "\n")
+
+        out.close()
+
+
+class CorpusReplaceOrthFromTxtJob(Job):
+    """
+    Merge raw text back into a bliss corpus
+    """
+
+    def __init__(self, corpus, text_file):
+        """
+
+        :param Path corpus: a bliss corpus file
+        :param Path text_file: a raw or gzipped text file
+        """
+        self.corpus_path = corpus
+        self.text_file = text_file
+
+        self.out_corpus = self.output_path("corpus.xml.gz")
+
+    def tasks(self):
+        yield Task('run', mini_task=True)
+
+    def run(self):
+        c = corpus.Corpus()
+        c.load(tk.uncached_path(self.corpus_path))
+
+        if tk.uncached_path(self.text_file).endswith(".gz"):
+            text = gzip.open(tk.uncached_path(self.text_file), "r")
+        else:
+            text = open(tk.uncached_path(self.text_file), "rt")
+
+        for recording in c.all_recordings():
```

Comment:
You can directly iterate over all segments and then do something like:
```suggestion
        for segment, line in itertools.zip_longest(c.segments(), f)
           assert segment is not None
           assert line is not None
```